### PR TITLE
Add feature for index mapping auto-put

### PR DESCRIPTION
**Pull request by thecoop**
Add a feature for nodes supporting `AutoPutMappingAction`

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticsearchmachine test this please

**Comment by thecoop** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 1]
```diff
@@ -207,6 +208,8 @@ public class IndicesService extends AbstractLifecycleComponent
         Setting.Property.NodeScope
     );
 
+    static final NodeFeature SUPPORTS_AUTO_PUT = new NodeFeature("indices.supports_auto_put");
```
Comment (ldematte) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java:211]: nit: to keep it in line with FeatureInfrastructureFeatures, I propose to change it to "indices.auto_put_supported"

**Review by ldematte — APPROVED** [review]
LGTM, with one minor comment

**Review by DaveCTurner — APPROVED** [review]
LGTM2 once Lorenzo's naming comment is addressed.

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/indices/IndicesService.java]: Good suggestion

**Pull request merged**